django/db/migrations/optimizer.py

```diff
@@ -167,9 +117,9 @@ def reduce(self, operation, other, in_between=None):
         Either returns a list of zero, one or two operations,
         or None, meaning this pair cannot be optimized.
         """
-        for ia, ib, om in self.reduce_methods:
-            if isinstance(operation, ia) and isinstance(other, ib):
-                return om(operation, other, in_between or [])
+        method = self.reduce_methods.get((type(operation), type(other)), None)
+        if method is not None:
```

Comment:
`None` as  a default value to `get()` is not required here. I would also just do `if method` since `None` is falsey it's really a nitpick.